src/crush/CrushWrapper.cc

```diff
@@ -2347,7 +2347,12 @@ int CrushWrapper::_choose_type_stack(
       underfull_buckets[j].insert(item);
     }
   }
-  ldout(cct, 20) << __func__ << " underfull_buckets " << underfull_buckets << dendl;
+  ldout(cct, 20) << __func__ << " underfull_buckets " << "[";
+  for (auto p = underfull_buckets.begin(); p != underfull_buckets.end(); ++p) {
```

Comment:
agreed! should compare the addresses of them if we want to go with the convoluted way.